tests/block/test_get_block.py

```diff
@@ -0,0 +1,269 @@
+    mock_get_block_by_id.return_value = block
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'block',
+        'get',
+        '--id',
+        EXISTING_BLOCK_IDENTIFIER,
+    ])
+
+    expected_result = {
+        'result': block.get('data'),
+    }
+
+    assert PASSED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert expected_result == json.loads(result.output)
+    assert EXISTING_BLOCK_IDENTIFIER == block.get('data').get('header_signature')
+
+
+def test_get_block_invalid_node_url():
+    """
+    Case: get a block by identifier by passing invalid node URL.
+    Expect: the following node URL is invalid error message.
+    """
+    invalid_node_url = 'domainwithoutextention'
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'block',
+        'get',
+        '--id',
+        EXISTING_BLOCK_IDENTIFIER,
+        '--node-url',
+        invalid_node_url,
+    ])
+
+    expected_error = {
+        'errors': {
+            'node_url': [
+                f'The following node URL `{invalid_node_url}` is invalid.',
+            ],
+        },
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error) in result.output
+
+
+@pytest.mark.parametrize('node_url_with_protocol', ['http://masternode.com', 'https://masternode.com'])
+def test_get_block_node_url_with_protocol(node_url_with_protocol):
+    """
+    Case: get a block by identifier by passing node URL with explicit protocol.
+    Expect: the following node URL contains protocol error message.
+    """
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'block',
+        'get',
+        '--id',
+        EXISTING_BLOCK_IDENTIFIER,
+        '--node-url',
+        node_url_with_protocol,
+    ])
+
+    expected_error = {
+        'errors': {
+            'node_url': [
+                f'Pass the following node URL `{node_url_with_protocol}` without protocol (http, https, etc.).',
+            ],
+        },
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error) in result.output
+
+
+def test_get_block_non_existing_node_url():
+    """
+    Case: get a block by identifier by passing non-existing node URL.
+    Expect: check if node running at URL error message.
+    """
+    non_existing_node_url = 'non-existing-node.com'
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'block',
+        'get',
+        '--id',
+        EXISTING_BLOCK_IDENTIFIER,
+        '--node-url',
+        non_existing_node_url,
+    ])
+
+    expected_error = {
+        'errors': f'Please check if your node running at http://{non_existing_node_url}:8080.',
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error) in result.output
+
+
+def test_get_block_non_existing_identifier():
```

Comment:
Please move this test, after this test `test_get_block_invalid_identifier`.

Reply:
@Alladin9393, could you propose any pattern of tests order which we can apply to any project we will do? For example, `valid functionality -> invalid functionality` or `step-by-step tests cases`, or? Because for now, we do not follow anyone.